File: 3.1P/Clock/Clock/Counter.cs
using System;


namespace ClockTask
{
  

   
    public class Counter
    {
        // Fields
        private int _count;
        private string _name;
        // Constructor
        public Counter(string name)
        {
            _name = name;
            _count = 0;
        }
        // Functions
        public void Increment()
        {
            _count++;
        }

        public void Reset()
        {
            _count = 0;
        }
       // Properties
        public string Name
        {
            get
            {
                return _name;
            }
            set
            {
                _name = value;
            }
        }
        
        public int Ticks
        {
            get
            {
                return _count;
            }
        }
    }
}

File: 11.1P/Counter.py
class Counter:
    def __init__(self, name):
        self._name = name
        self._count = 0
    
    def Increment(self):
        self._count += 1
    
    def Reset(self):
        self._count = 0
    
    @property
    def Name(self):
        return self._name
    
    @Name.setter
    def Name(self, value):
        self._name = value
    
    @property
    def Ticks(self):
        return self._count
